tests/org.eclipse.n4js.n4mf.xpect.ui.tests/src/org/eclipse/n4js/n4mf/tests/N4MFAvoidRefreshDocumentProviderTest.xtend

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.n4mf.tests
+
+import java.lang.reflect.Field
+import org.eclipse.core.internal.resources.Workspace
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.ResourcesPlugin
+import org.eclipse.n4js.N4JSUiInjectorProvider
+import org.eclipse.n4js.tests.helper.documentprovider.CountPostChangeBroadcastChangeNotificationManager
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
+import org.eclipse.xtext.ui.XtextProjectHelper
+import org.eclipse.xtext.ui.testing.AbstractEditorTest
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import static java.util.UUID.randomUUID
+
+import static extension org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil.*
+
+/**
+ * This class tests the fix for refresh file problem in XtextDocumentProvider. GH-270.
+ */
+@RunWith(XtextRunner)
+@InjectWith(N4JSUiInjectorProvider)
+class N4MFAvoidRefreshDocumentProviderTest extends AbstractEditorTest {
+
+	val N4MF_EDITOR_ID = 'org.eclipse.n4js.n4mf.N4MF'
+	val PROJECT_NAME = 'testProject'
+	val MF_FILE = 'manifest.nfmf'
+	val SRC = 'src';
+	
+	var IFile n4mfFile;
+
+	override setUp() throws Exception {
+		super.setUp()
+		createN4JSProjectWithXtextNature
+	}
+
+	@Test
+	public def void noRefreshWhenOpenningN4MFFileTest() {
+		val content = 'class C {}';		
+		createFileWithContent(content)
+		val workspace = ResourcesPlugin.getWorkspace() as Workspace
+
+		val countBroadcastChangeNotificationManager = new CountPostChangeBroadcastChangeNotificationManager(workspace)
+
+		// Use reflection to replace workspace's notification manager with our custom notification manager
+		var Field notificationManagerField
+		notificationManagerField = Workspace.getDeclaredField("notificationManager")
+		notificationManagerField.accessible = true
+		notificationManagerField.set(workspace, countBroadcastChangeNotificationManager)
+		// We need to startup the notification so that it can get the initial state of the workspace
+		countBroadcastChangeNotificationManager.startup(null)
```

Comment:
We certainly should reset the field value in a finally block to the previous value

Reply:
Done

Reply:
Also, I am using ReflectExtensions for reflection now.